Sources/NIO/Bootstrap.swift

```diff
@@ -675,27 +675,8 @@ public final class DatagramBootstrap {
 /* for tests */ internal struct ChannelOptionStorage {
```

Comment:
Here's a question: should this object be public? It's annoying as hell to have to reimplement this in other NIO channel implementations, especially as it's quite fussy.

Reply:
I'm happy for us to just file an issue to make it public and document it and clean it up, rather than do it in this PR, of course.

Reply:
great idea, adding it as `ChannelOptions.Storage`, nicely nested too

Reply:
@Lukasa oh sorry, I put it in this PR, okay with that or split?

Reply:
Sure, it's fine here.

Reply:
I've split it in the end.